mlir/include/mlir/Pass/Pass.h

```diff
@@ -94,6 +94,7 @@ class Pass {
     Option(Pass &parent, StringRef arg, Args &&...args)
         : detail::PassOptions::Option<DataType, OptionParser>(
               parent.passOptions, arg, std::forward<Args>(args)...) {}
+    Option &operator=(const Option &other) = default; // gcc11 -Wdeprecated-copy
```

Comment:
Don't need this comment?

Reply:
@dwblaikie had asked for specific comment about why this was added.

Reply:
I think I said that the previous comment wasn't clear and it could/should be clearer. Which I guess is still true.

Sounds like you/we don't know why this code is needed, and it isn't needed on more recent GCCs? Another case of "maybe we should disable the warning due to a false positive"?

Reply:
See above, done.

Reply:
It does not need to exist if not for the warning...

Reply:
> See above, done.

What was done exactly? Sorry, I'm not quite following.

Re: Doesn't need a warning based justification: I'm concerned someone would remove it since it's unneeded, and break the build again. Or just be confused why it's there if it's not necessary/obvious why it would be needed - that it might be trying to communicate something else that's not clear/confusing.

Reply:
> > See above, done.
> 
> What was done exactly? Sorry, I'm not quite following.

There were 2 discussions on the same topic.
I removed the explicit operator from the code and disabled the warning for gcc versions <12 if flag is supported.
The cmake files also clarifies why the warning is disabled.
This will keep the compiler warning for newer versions which do not report the false positive without adding unneeded code.